tests/NHSD.BuyingCatalouge.Ordering.Api.Testing.Data/EntityBuilder/OrderEntityBuilder.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.Entities;
+
+namespace NHSD.BuyingCatalouge.Ordering.Api.Testing.Data.EntityBuilder
+{
+    public sealed class OrderEntityBuilder
+    {
+        private readonly OrderEntity _orderEntity;
+
+        public static OrderEntityBuilder Create()
+        {
+            return new OrderEntityBuilder();
+        }
+
+        private OrderEntityBuilder()
+        {
+            _orderEntity = new OrderEntity()
+            {
+                Description = "Some Description",
+                OrganisationId = Guid.NewGuid(),
+                OrderStatusId = 0,
+                LastUpdated = DateTime.UtcNow,
+                LastUpdatedBy = Guid.NewGuid(),
+                Created = DateTime.UtcNow
+            };
+        }
+
+        public OrderEntityBuilder WithOrderId(string orderId)
+        {
+            _orderEntity.OrderId = orderId;
+            return this;
+        }
+
+        public OrderEntityBuilder WithDescription(string description)
+        {
+            _orderEntity.Description = description;
+            return this;
+        }
+
+        public OrderEntityBuilder WithOrderStatusId(int statusId)
+        {
+            _orderEntity.OrderStatusId = statusId;
+            return this;
+        }
+
+        public OrderEntityBuilder WithLastUpdated(DateTime lastUpdated)
+        {
+            _orderEntity.LastUpdated = lastUpdated;
+            return this;
+        }
+
+        public OrderEntityBuilder WithLastUpdatedBy(Guid lastUpdatedBy)
+        {
+            _orderEntity.LastUpdatedBy = lastUpdatedBy;
+            return this;
+        }
+
+        public OrderEntityBuilder WithDateCreated(DateTime dateCreated)
+        {
+            _orderEntity.Created = dateCreated;
+            return this;
+        }
+
+        public OrderEntity Build()
+        {
+            return _orderEntity;
```

Comment:
This is a change in style to the builder pattern used in the other API projects. I would have expected to see the creation of the object at the time of building in this method. Not that I can think of anything wrong with it just different.

Reply:
👍